src/serde_impl.rs

```diff
@@ -93,16 +93,22 @@ macro_rules! impl_serde_for_symbol {
                 &self,
                 serializer: T,
             ) -> ::core::result::Result<T::Ok, T::Error> {
-                self.value.serialize(serializer)
+                // When the value is deserialized below, it will be converted back to a non-zero
+                // type by adding 1 to the value, so we need to account for that here by subtracting
+                // 1 from the value.
+                let zeroable_value = self.value.get() - 1;
+                zeroable_value.serialize(serializer)
```

Comment:
This should use `Symbol::to_usize` which internally performs the `-1` if necessary. There are symbols that don't need this, e.g. `usize` can be a `Symbol` and for it, performing `-1` would be incorrect.

Reply:
Ah, I didn't spot that Symbol is implemented for usize directly, and not just for SymbolUsize. Good point, will fix.